### PR TITLE
Add destination folder

### PR DESCRIPTION
**Pull request by jawrainey**

### CODE REVIEW

File: csv2docx/cli.py [hunk 1]
```diff
@@ -1,7 +1,22 @@
 from mailmerge import MailMerge
 import click
 import csv
-import sys
+import pathlib
+
+
+def create_path_if_not_exists(path: str) -> pathlib.Path:
```
Comment (jawrainey) [inline csv2docx/cli.py:7]: Not a fan of the name of this method ... open to suggestions

File: csv2docx/cli.py [hunk 2]
```diff
@@ -15,9 +30,13 @@
 @click.option(
     '--delimiter', '-d',
     default=";",
-    help='delimiter used in your csv. Default is \';\'')
-def convert(data, template, delimiter):
-    print ("Getting .docx template and .csv data files ...")
+    help='Delimiter used in your csv. Default is \';\'')
+@click.option(
+    '--path', '-p',
```
Comment (jawrainey) [inline csv2docx/cli.py:35]: Maybe destination is a better name?

**Comment by jawrainey** [comment]
To test run:

    poetry run convert -t tests/data/example.docx -c tests/data/example.csv

Then throw whatever you can think at it:

Nested folders:

    poetry run convert -t tests/data/example.docx -c tests/data/example.csv -p dave/jay/output

Shortcuts:

    poetry run convert -t tests/data/example.docx -c tests/data/example.csv -p ~/Desktop/output

Relative:

    poetry run convert -t tests/data/example.docx -c tests/data/example.csv -p ./output

IMHO, we should not merge this until we have written tests to validate the above and whatever other edge cases we can think of. This is my first time using pathlib, so there may be gotchas I am unaware of, but it seems pretty neat.

File: csv2docx/cli.py (hunk 1 above)
Comment (jawrainey) [reply csv2docx/cli.py:7]: Also, the problem is that the parameter `path` can be a directory `output`, an absolute path `/user/bin/output` or relative path `../../output` and we should except all those conditions.

Rather than deal with edge cases, I used pathlib as it converts the given path to a `concrete path` that is user platform specific 👍

File: csv2docx/cli.py [hunk 3]
```diff
@@ -32,15 +51,16 @@ def convert(data, template, delimiter):
         # see if all fields are accounted for in the .csv header
         column_in_data = set(docx_mergefields) - set(csv_headers)
         if len(column_in_data) > 0:
-            print (f"{column_in_data} is in the word document, but not csv.")
+            print(f"{column_in_data} is in the word document, but not csv.")
             return
 
         print("All fields are present in your csv. Generating Word docs ...")
 
+        path = create_path_if_not_exists(path)
```
Comment (jawrainey) [inline csv2docx/cli.py:59]: As above, but to reiterate: can you think of a better name?

File: csv2docx/cli.py [hunk 4]
```diff
@@ -32,15 +51,16 @@ def convert(data, template, delimiter):
...
+
         for counter, row in enumerate(csvdict):
             # Must create a new MailMerge for each file
             docx = MailMerge(template)
-            single_document = {key : row[key] for key in docx_mergefields}
+            single_document = {key: row[key] for key in docx_mergefields}
             docx.merge_templates([single_document], separator='page_break')
-            # TODO: write to user-defined subfolder
-            docx.write(f"{counter}.docx")
+            docx.write(f"{path}/{counter}.docx")
```
Comment (jawrainey) [inline csv2docx/cli.py:66]: Just had a thought that this might cause issues on windows given we are using `/`. However, because we convert `path` to type `pathlib.Path` it means that once we reach this `docx.write` that it is now a pathlib.path and not a string. This is _very important_ because if you use the backslash operator (`/`) after a pathlib.path object then it will join them 👍

File: csv2docx/cli.py (hunk 1 above)
Comment (davidverweij) [reply csv2docx/cli.py:7]: Perhaps go with 
`def concrete_path(path: str) -> pathlib.Path:`
with a note that this method will create the directory if non-existent. 

Ps. does it handle a nested non-existent directory?

File: csv2docx/cli.py (hunk 2 above)
Comment (davidverweij) [reply csv2docx/cli.py:35]: I would go with destination yes

File: csv2docx/cli.py (hunk 4 above)
Comment (davidverweij) [reply csv2docx/cli.py:66]: So that means the "/" will work in this case right? Also, the counter would ideally be a filename based on the .csv - see also #16 - but that might be for another PR

File: csv2docx/cli.py (hunk 1 above)
Comment (jawrainey) [reply csv2docx/cli.py:7]: Great suggestion re:name 👍 

Yes -- it handles nested non-existent directories thanks to the `parents` option.

File: csv2docx/cli.py (hunk 4 above)
Comment (jawrainey) [reply csv2docx/cli.py:66]: Yes, it would work in this case, but I feel that we should probably instead create a variable before adding it to `docx.write` to make it more explicit, e.g. `write(f"{path}.docx")` where path is defined before 👍

File: csv2docx/cli.py (hunk 2 above)
Comment (jawrainey) [reply csv2docx/cli.py:35]: @davidverweij -- one minor issue with destination is that the `data` option has the `-d` flag ...

File: csv2docx/cli.py (hunk 1 above)
Comment (jawrainey) [reply csv2docx/cli.py:7]: How about `absolute_path` as it's more common terminology?

**Comment by jawrainey** [comment]
@davidverweij -- I have made the changes you've suggested above and merged master into this PR. Could you test it and let me know your thoughts? Once happy, I can write unit tests for the method (or all methods?!) to put the infrastructure in place for #11, #12 , etc prior to merging this with master. Tests in place will make validating future PR/changes easier.

**Comment by davidverweij** [comment]
Pulled into 'output_destination' branch to work on further. Please `--set-upstream-to origin/output_destination` to track updates

**Comment by davidverweij** [comment]
Continuing this on the output branch, see PR #18

File: csv2docx/cli.py (hunk 2 above)
Comment (salmannotkhan) [reply csv2docx/cli.py]: i would go with -o output it feels standard
